diff --git a/go.mod b/go.mod
index e2ab3786..bcfaacd8 100644
--- a/go.mod
+++ b/go.mod
@@ -50,5 +50,6 @@ require (
 	golang.org/x/text v0.3.4 // indirect
 	google.golang.org/grpc v1.37.0
 	google.golang.org/protobuf v1.26.0
+	gopkg.in/yaml.v2 v2.4.0 // indirect
 	gotest.tools v2.2.0+incompatible
 )
diff --git a/go.sum b/go.sum
index 6f381edf..ac360199 100644
--- a/go.sum
+++ b/go.sum
@@ -519,8 +519,9 @@ gopkg.in/yaml.v2 v2.2.1/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.2/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.4/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
 gopkg.in/yaml.v2 v2.2.5/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
-gopkg.in/yaml.v2 v2.3.0 h1:clyUAQHOM3G0M3f5vQj7LuJrETvjVot3Z5el9nffUtU=
 gopkg.in/yaml.v2 v2.3.0/go.mod h1:hI93XBmqTisBFMUTm0b8Fm+jr3Dg1NNxqwp+5A1VGuI=
+gopkg.in/yaml.v2 v2.4.0 h1:D8xgwECY7CYvx+Y2n4sBz93Jn9JRvxdiyyo8CTfuKaY=
+gopkg.in/yaml.v2 v2.4.0/go.mod h1:RDklbk79AGWmwhnvt/jBztapEOGDOx6ZbXqjP6csGnQ=
 gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c h1:dUUwHk2QECo/6vqA44rthZ8ie2QXMNeKRTHCNY2nXvo=
 gopkg.in/yaml.v3 v3.0.0-20200313102051-9f266ea9e77c/go.mod h1:K4uyk7z7BCEPqu6E+C64Yfv1cQ7kz7rIZviUmN+EgEM=
 gotest.tools v2.2.0+incompatible h1:VsBPFP1AI068pPrMxtb/S8Zkgf9xEmTLJjfM+P5UIEo=
diff --git a/vms/platformvm/proposal_block.go b/vms/platformvm/proposal_block.go
index 45b097c7..abe02aaa 100644
--- a/vms/platformvm/proposal_block.go
+++ b/vms/platformvm/proposal_block.go
@@ -172,6 +172,9 @@ func (pb *ProposalBlock) Verify() error {
 	parentState := parent.onAccept()
 
 	var err TxError
+	if _, ok := tx.(*UnsignedRewardValidatorTx); ok {
+		pb.vm.ctx.Log.Info("csv ,,,,,,# block,%d\n", pb.Height())
+	}
 	pb.onCommitState, pb.onAbortState, pb.onCommitFunc, pb.onAbortFunc, err = tx.SemanticVerify(pb.vm, parentState, &pb.Tx)
 	if err != nil {
 		txID := tx.ID()
diff --git a/vms/platformvm/reward_validator_tx.go b/vms/platformvm/reward_validator_tx.go
index 44fee902..8ef26e3f 100644
--- a/vms/platformvm/reward_validator_tx.go
+++ b/vms/platformvm/reward_validator_tx.go
@@ -12,7 +12,9 @@ import (
 	"github.com/ava-labs/avalanchego/ids"
 	"github.com/ava-labs/avalanchego/vms/components/avax"
 	"github.com/ava-labs/avalanchego/vms/components/verify"
+	"github.com/ava-labs/avalanchego/vms/secp256k1fx"
 
+	"github.com/ava-labs/avalanchego/utils/constants"
 	safemath "github.com/ava-labs/avalanchego/utils/math"
 )
 
@@ -43,6 +45,12 @@ type UnsignedRewardValidatorTx struct {
 	shouldPreferCommit bool
 }
 
+func getZulu(t time.Time) string {
+	loc, _ := time.LoadLocation("UTC")
+	t = t.In(loc)
+	return t.Format(time.RFC3339)
+}
+
 // SemanticVerify this transaction performs a valid state transition.
 //
 // The current validating set must have at least one member.
@@ -147,6 +155,11 @@ func (tx *UnsignedRewardValidatorTx) SemanticVerify(
 		// Provide the reward here
 		if stakerReward > 0 {
 			outIntf, err := vm.fx.CreateOutput(stakerReward, uStakerTx.RewardsOwner)
+			txID := uStakerTx.ID().String()
+			owner, _ := uStakerTx.RewardsOwner.(*secp256k1fx.OutputOwners)
+			addr, _ := vm.FormatLocalAddress(owner.Addrs[0])
+			nodeID := uStakerTx.Validator.ID().PrefixedString(constants.NodeIDPrefix)
+			vm.ctx.Log.Info("csv %s,%s,staking,%d,%s,%s,,\n", txID, addr, stakerReward, nodeID, getZulu(currentTime))
 			if err != nil {
 				return nil, nil, nil, nil, permError{
 					fmt.Errorf("failed to create output: %w", err),
@@ -217,6 +230,11 @@ func (tx *UnsignedRewardValidatorTx) SemanticVerify(
 		// Reward the delegator here
 		if delegatorReward > 0 {
 			outIntf, err := vm.fx.CreateOutput(delegatorReward, uStakerTx.RewardsOwner)
+			txID := uStakerTx.ID().String()
+			owner, _ := uStakerTx.RewardsOwner.(*secp256k1fx.OutputOwners)
+			addr, _ := vm.FormatLocalAddress(owner.Addrs[0])
+			nodeID := uStakerTx.Validator.ID().PrefixedString(constants.NodeIDPrefix)
+			vm.ctx.Log.Info("csv %s,%s,delegator,%d,%s,%s,,\n", txID, addr, delegatorReward, nodeID, getZulu(currentTime))
 			if err != nil {
 				return nil, nil, nil, nil, permError{
 					fmt.Errorf("failed to create output: %w", err),
@@ -244,6 +262,11 @@ func (tx *UnsignedRewardValidatorTx) SemanticVerify(
 		// Reward the delegatee here
 		if delegateeReward > 0 {
 			outIntf, err := vm.fx.CreateOutput(delegateeReward, vdrTx.RewardsOwner)
+			txID := uStakerTx.ID().String()
+			owner, _ := vdrTx.RewardsOwner.(*secp256k1fx.OutputOwners)
+			addr, _ := vm.FormatLocalAddress(owner.Addrs[0])
+			nodeID := uStakerTx.Validator.ID().PrefixedString(constants.NodeIDPrefix)
+			vm.ctx.Log.Info("csv %s,%s,delegatee,%d,%s,%s,,\n", txID, addr, delegateeReward, nodeID, getZulu(currentTime))
 			if err != nil {
 				return nil, nil, nil, nil, permError{
 					fmt.Errorf("failed to create output: %w", err),
